### ISSUE TITLE
Unstated dependencies found when building in minimal (docker) environments

### ISSUE THREAD

**Issue by gcoxmoz** [mozilla/mig#308]
I was trying to bundle up a Dockerfile to make mig happen.
I hit a few snags; you might ignore 1-2, but I think 3 and 4 are more actionable:

```
FROM ubuntu:16.04
ENV GOPATH=/go
RUN apt-get update \
 && apt-get --no-install-recommends --no-install-suggests -y install \
    git \
    golang \
  && rm -rf /var/lib/apt/lists/*

WORKDIR $GOPATH
RUN go get mig.ninja/mig
WORKDIR $GOPATH/src/mig.ninja/mig
RUN bash tools/standalone_install.sh
```
This pretty much matches the install instructions, except:

1. This doesn't include ca-certificates, so 'go get' gripes:

`package mig.ninja/mig: unrecognized import path "mig.ninja/mig" (https fetch: Get https://mig.ninja/mig?go-get=1: x509: failed to load system roots and no roots provided)`

Now, you can blame me for this if you want, as it's a suggested package and I did --no-install-suggests.  But I include it because it was the first bump, and leads me to...

2. If I include ca-certificates, it fetches, but then bombs out during install because of the lack of sudo plus other packages:

```
tools/standalone_install.sh: line 42: sudo: command not found
missing packages:  libreadline-dev rabbitmq-server postgresql mercurial make gcc tmux curl rng-tools
would you like to install the missing packages? (need sudo) y/n> tools/standalone_install.sh: line 92: [: =: unary operator expected
```

So since it wants to be interactive, I have to fold all of those packages into the Dockerfile ahead of time.  No problem, I include everything listed.

3. The build then fails with

```
# mig.ninja/mig/vendor/github.com/bobappleyard/readline
vendor/github.com/bobappleyard/readline/readline.go:41:19: fatal error: stdio.h: No such file or directory
compilation terminated.
Makefile:147: recipe for target 'mig-console' failed
```

This vendor patch assumes a dependency of libc6-dev, not asked for in standalone_install.sh.

4.  Further build attempts fail because it's not a fully-functional environment:

```
psql: could not connect to server: No such file or directory
	Is the server running locally and accepting
	connections on Unix domain socket "/var/run/postgresql/.s.PGSQL.5432"?
```

If I flip my final RUN command to be a simple 'make', it completes.  But that means not running standalone_install.sh at all, and thus not discovering all of the requirements I needed to get this far.

Basically, "the requirements-gathering here is kinda messy," is what I'm trying to say.

**Comment by ameihm0912** [comment]
@gcoxmoz https://github.com/mozilla/mig/blob/master/Dockerfile should help out a bit with this now; this should stay up to date as it is now used in our Travis builds.
